bin/pytorch_inference/unittest/CCommandParserTest.cc

```diff
@@ -288,4 +242,58 @@ BOOST_AUTO_TEST_CASE(testRequestHandlerExitsLoop) {
     BOOST_REQUIRE_EQUAL(1, parsed.size());
 }
 
+BOOST_AUTO_TEST_CASE(testParsingBatch) {
```

Comment:
I think a test where you run inference for randomised strings of tokens one at a time and in batch and check you always get the same results would be good. This way you can test many more cases.

Reply:
The test suite is only testing the input parsing code. `evaluate.py` and the `test_run.json` files cover the case of asserting batch inputs give the same results as a single input. Unfortunately we can't randomise those inputs yet.

Reply:
Perhaps something to revisit in a separate PR. I'm happy for this to go in without the extra testing.